### PR TITLE
[TASK] Security documentation

### PR DESCRIPTION
**Pull request by kdambekalns**
This documents node privileges and their matchers, as well as how to restrict
access to backend modules.

### CODE REVIEW

**Comment by kdambekalns** [comment]
Hi @bwaidelich and @foerthner, could you have a look?

File: TYPO3.Neos/Documentation/CreatingASite/Security.rst [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+A privilege that targets editing of node properties.
+
+Usage example:
+
+.. code-block:: yaml
+
+  privilegeTargets:
+    'TYPO3\\TYPO3CR\\Security\\Authorization\\Privilege\\Node\\EditNodePropertyPrivilege':
+      'Some.Package:SomeIdentifier':
+        matcher: >-
```
Comment (bwaidelich) [inline TYPO3.Neos/Documentation/CreatingASite/Security.rst]: I guess this will be rendered correctly?

**Comment by bwaidelich** [comment]
@kdambekalns how awesome is this? how awesome are you?!
:+1: by (quick) reading
It might be worth mentioning that this is a *blacklist* by default, so that the privilege won't match if one of the conditions don't match. So the example:
```
  privilegeTargets:
    'TYPO3\\TYPO3CR\\Security\\Authorization\\Privilege\\Node\\CreateNodePrivilege':
      'Some.Package:SomeIdentifier':
        matcher: isDescendantNodeOf("c1e528e2-b495-0622-e71c-f826614ef287") && createdNodeIsOfType("TYPO3.Neos.NodeTypes:Text")
```
will actually only affect nodes of that type (and subtypes). All users will still be able to create other node types.
Unless you also add a more generic privilege target
```
  privilegeTargets:
    'TYPO3\\TYPO3CR\\Security\\Authorization\\Privilege\\Node\\CreateNodePrivilege':
      'Some.Package:SomeIdentifier':
        matcher: isDescendantNodeOf("c1e528e2-b495-0622-e71c-f826614ef287")
```
that will be abstained by default.
It's the same with MethodPrivileges, but with those we abstain all actions by default (in Neos that is)

File: TYPO3.Neos/Documentation/CreatingASite/Security.rst (hunk 1 above)
Comment (kdambekalns) [inline TYPO3.Neos/Documentation/CreatingASite/Security.rst]: Yes.

**Comment by kdambekalns** [comment]
Thanks, @bwaidelich - it might make sense to provide a somewhat bigger, "useful" annotated example, I guess.

**Comment by skurfuerst** [comment]
big fat :+1: from my side!

**Comment by kdambekalns** [comment]
@bwaidelich Added your note about blacklisting behaviour.

**Comment by foerthner** [comment]
Really nice, thanks so much! For me this can go in. The section about modules might change again as soon as we have a real module privilege type. However, for now it's like described, so fine with me :)

**Comment by bwaidelich** [comment]
Awesome, :+1:

**Pull request merged**